### DISCUSSION TITLE
MiniDexed on "normal" OS

### DISCUSSION THREAD

**Discussion by grbfst** [probonopd/MiniDexed#598] [Q&A] [answered]
I'm building a touch oriented instrument with a Raspberry pi for someone who works with the (demented) elderly. I'm using a MCP3008 (or maybe two) to convert touch to midi. Since this is running on python I'm wondering if there is a way to run MiniDexed without the bare metal but with the LCD screen and rotary encoder, so on Raspberry Pi OS lite. 
If so, I can run the touch to midi python program on the same Raspberry,

**Comment by probonopd** [comment, marked as answer]
Hi @grbfst.

Since MiniDexed is based on Circle and there is currently no way to run Circle applications on Linux, this is not possible at this time unfortunately.

You may achieve a similar solution by running the Dexed standalone application on Raspberri Pi OS, or by running the Dexed VSTi inside a VST host.

**Comment by grbfst** [comment]
I have tried to compile Dexed  standalone on a Raspberry 3 and 4 but every time it gets no further than "building juceaide". I know Dexed is another repository but I've seen you around there so maybe you might give some pointers?

**Comment by stanleygarvey** [comment]
Hi,
try git branch dda2e72.
This complied for me on RPI 4 running bookworm OS  and previous bullseye OS.
I know there are problems compiling current branch, I can't remember what though... Should have documented it.
Hope that helps,
Regards.

18 Jan 2024 09:17:40 grbfst ***@***.***>:

**Comment by stanleygarvey** [comment]
Okay I am rebuilding the branch, first gotcha is Projucer binary is X86-64 and needs recomp to aarch64.
I am starting to recall the the journey to hell it is to build this...

18 Jan 2024 17:33:36 Stanley Garvey ***@***.***>:

**Comment by probonopd** [comment]
If you want to build Dexed, the appropriate place to discuss that would be https://github.com/asb2m10/dexed/discussions - this repository here is about MiniDexed.

**Comment by pik33** [comment]
MiniDexed is based on Circle, that's C++ bare metal environment. Porting is of course possible, but it is not the work that can be done "automatically". While the synth code itself will be the same, near all the I/O stuff has to be rewritten

**Comment by grbfst** [comment]
I recently compiled on a Raspberry pi 5 and it went without trouble (and fast!). Copied the file to a RPI4 and it ran just fine (don't forget to copy the icon file too). On a RPI3 just doesn't compile at all and compiling on a 4 just hangs.
Just so you know.

**Reply by probonopd** [reply]
MiniDexed currently does not run on Raspberry Pi 5 yet, but possibly will at some time if Circle supports it.
We are cross-compiling it for the other Raspberry Pi models from the Pi Zero to the Pi 4.
